msmtools/analysis/dense/gpcca.py

```diff
@@ -1169,10 +1182,10 @@ def minChi(self, m_min, m_max):
             minChi_list.append(minChi)
             
         return minChi_list
-        
-        
+
     # G-PCCA coarse-graining   
-    def optimize(self, m, full_output=False):
+    def optimize(self, m: Union[int, Tuple[int, int], List[int], Dict[str, int]],
```

Comment:
Do we really need so much input type possibilities here? I think this creates unnecessary complexity and redundancy. int and dict are already enough. Mb one could add a tuple restricted to two items (Tuple[int, int] is exactly this, right?), but why would one need a list of possibly more than two elements?